packages/filigran-ui/src/components/clients/file-input.tsx

```diff
@@ -0,0 +1,112 @@
+'use client'
+import {AddNotesIcon, ReadMoreIcon, TaskIcon} from 'filigran-icon'
+import {useState} from 'react'
+import {cn, fixedForwardRef} from '../../lib/utils'
+
+const FileStatusDisplay = ({
+  icon: Icon,
+  text,
+}: {
+  icon: React.ElementType
+  text: string
+}) => (
+  <div className="flex flex-col items-center justify-center pointer-events-none text-gray-700 dark:text-gray-300">
+    <Icon className="h-10 w-10" />
+    <p className="mt-s text-sm flex items-center justify-center text-center">
+      {text}
+    </p>
+  </div>
+)
+
+interface FileInputProps {
+  texts: Partial<InputText>
+  className?: string | undefined
+  handleFileChange: (file) => void
+}
+
+interface InputText {
+  dragActive: string
+  dragUnactive: string
+  fileOk?: string
+}
+
+const defaultTexts: InputText = {
+  dragActive: 'Drop the file here',
+  dragUnactive: 'Select a file',
+}
+
+function GenericFileInput(
+  {texts, className, handleFileChange, ...props}: FileInputProps,
```

Comment:
si jamais il gueule parce que le texts peut être undefined tu peux faire :

```suggestion
  {texts = {}, className, handleFileChange, ...props}: FileInputProps,
```